R/measures.R

```diff
@@ -1415,27 +1415,25 @@ iauc.uno = makeMeasure(id = "iauc.uno", minimize = FALSE, best = 1, worst = 0,
 #' @rdname measures
 #' @format none
 ibrier = makeMeasure(id = "ibrier", minimize = TRUE, best = 0, worst = 1,
-  properties = c("surv", "req.truth", "req.model", "req.task"),
+  properties = c("surv", "req.prob", "req.feats"),
   name = "Integrated brier score using Kaplan-Meier estimator for weighting",
   note = "Only works for methods for which probabilities are provided via pec::predictSurvProb. Currently these are only coxph and randomForestSRC. To set an upper time limit, set argument max.time (defaults to max time in test data). Implemented in pec::pec",
   fun = function(task, model, pred, feats, extra.args) {
     requirePackages(c("survival", "pec"))
-    targets = getTaskTargets(task)
-    tn = getTaskTargetNames(task)
-    f = as.formula(sprintf("Surv(%s, %s) ~ 1", tn[1L], tn[2L]))
-    newdata = getTaskData(task)[model$subset, ]
-    max.time = extra.args$max.time %??% max(newdata[[tn[1L]]])
-    grid = seq(0, max.time, length.out = extra.args$resolution)
-
-    probs = predictSurvProb(getLearnerModel(model, more.unwrap = TRUE), newdata = newdata, times = grid)
-    perror = pec(probs, f, data = newdata[, tn], times = grid, exact = FALSE, exactness = 99L,
-      maxtime = max.time, verbose = FALSE, reference = FALSE)
-    
-
-    # FIXME: this might be the wrong number!
-    crps(perror, times = max.time)[1L, ]
-  },
-  extra.args = list(max.time = NULL, resolution = 1000L)
+    truth = getPredictionTruth(pred)
+    target = getPredictionTaskDesc(pred)$target
+    newdata = cbind(truth[,1], pred$data$truth.event, feats)
+    colnames(newdata)[1:2] = target
+    probs = getPredictionProbabilities(pred)
+    colnames(probs) = times.train = as.numeric(substr(colnames(probs), 11, 100))
+    times.test = truth[truth[,2] == 1,1]
+    grid = c(sort(unique(c(times.test))))
+    prob_columns = sapply(grid, function(t) max(which(t >= times.train - 10^(-13))))
```

Comment:
What's going on here? How does the training time relate to what column names to select?

Reply:
This is specific to the pec interface. We need the probabilities to the specific time points given by `times.test`. The probabilities only change from point to point in `times.train` ...

Reply:
Can you give an example of what this does please?

Reply:
Oh wait, you're talking about the survival times, not the time it took to train/test the model, right? Still not sure what's going on though.

Reply:
Yes, I talk about survival times. As the probabilities do not change from "training timepoint to training timepoint" we jus take the probabilities of the specific training time points. For understanding this you have to know some basics of the survival models. The time dependent part is usually not dependent on the variables (see e.g. $\lambda_0(t)$ in the proportional hazards model https://en.wikipedia.org/wiki/Proportional_hazards_model) that's why it can be handled like that.